extern/sector-storage/stores/remote.go

```diff
@@ -276,12 +308,57 @@ func (r *Remote) fetch(ctx context.Context, url, outname string) error {
 	case "application/x-tar":
 		return tarutil.ExtractTar(resp.Body, outname)
 	case "application/octet-stream":
-		return files.WriteTo(files.NewReaderFile(resp.Body), outname)
+		return WriteWithSize(files.NewReaderFile(resp.Body), outname, iosize)
```

Comment:
This used the files api to make things simpler. Instead of duplicating the code below, we can just do
```suggestion
		f, err := os.Create(outname)
		defer f.Close()
		if err != nil {
			return err
		}
		_, err = io.CopyBuffer(f, resp.Body, make([]byte, iosize))
		if err != nil {
			return err
		}
		return nil
```